books/projects/shnf/top.lisp

```diff
@@ -427,19 +456,20 @@
                (norm-pow j (norm-mul r (norm-pop 1 q)) 0))))               ;r * q * z^j
 
 (defund norm-mul (x y)
-  (declare (xargs :measure (+ (shf-count x) (shf-count y))))
-  (and (shfp x)
-       (if (atom x)
-           (mul-int x y)
-         (if (atom y)
-             (mul-int y x)
-           (case (car x)
-             (pop (case (car y)
-                    (pop (mul-pop-pop x y))
-                    (pow (mul-pop-pow x y))))
-             (pow (case (car y)
-                    (pop (mul-pop-pow y x))
-                    (pow (mul-pow-pow x y)))))))))
+  (declare (xargs :measure (+ (shf-count x) (shf-count y))
```

Comment:
Remove `shfp` test. In program mode it follow from guard. In logic mode is is not essential.